runtime/wasm/src/module/mod.rs

```diff
@@ -207,8 +207,10 @@ impl<C: Blockchain> WasmInstance<C> {
         let handler_name = trigger.handler_name().to_owned();
         let gas = self.gas.clone();
         let logging_extras = trigger.logging_extras().cheap_clone();
+        let error_context = trigger.trigger.error_context();
```

Comment:
This will create a `String` every time a handler is invoked, which is a lot; we should try to avoid this allocation overhead on every invocation. 

One way to do that is to pass a `&dyn MappingTriggerTrait` into `invoke_handler` instead of the `error_context: Option<String>` and then call `error_context()` only if an error actually happens.

Reply:
Looking some more into this, the fact that `to_asc_ptr` takes ownership of the trigger makes what I said above impossible; it's not clear to me that `to_asc_ptr` has to take ownership, but changing that would be quite a bit more work. So I think we should go with this as-is, and plan on optimizing this later on.